### ISSUE TITLE
Duplicate JsonProperty "source" in StripeCustomerUpdateOptions

### ISSUE THREAD

**Issue by xalhera** [stripe/stripe-dotnet#1181]
Hi,

I'm using a RESTFUL API to update customers data.
The C# model binder throws an error saying that there is a duplicate "source" JsonProperty in the class StripeCustomerUpdateOptions.

I solved this issue by providing my own class StripeCustomerUpdateOptions, but I want to alert you about that.

![image](https://user-images.githubusercontent.com/20375763/39856695-9836db0e-547c-11e8-8697-9c8ac816ef10.png)

**OS :** Windows 10 Entreprise (Version 16299.431)
**IDE :** Visual Studio 2017 (Version 15.7.1)
**Framework :** .NET Core 2.1.200
**Stripe.net :** 15.8.0

**Comment by remi-stripe** [comment]
@xalhera The idea here is that only the property set should be sent. At least it works with our library as is. If you do this: 
```
            StripeCustomer updated = new StripeCustomerService().Update(
              "cus_1234",
              new StripeCustomerUpdateOptions() {
                SourceToken = "tok_visa",
              }
            );
```

Only `source=tok_visa` is sent. The other one is not set so it's not sent.

Can you clarify what causes the issue specifically on your end? Is your approach trying to send both parameters or is it validating the properties at the upper level and blocking before even sending the request?

**Comment by xalhera** [comment]
Thanks for your reply, indeed this is the way I'm doing this in the end. Maybe I was not clear enough and I apologize for the inconvenience.

This is not an issue with the logic behind how to use your library but with the duplication of JSON properties.

For instance, you named these properties differently in C# because the compiler do not allow duplicate name properties.
Maybe I'm missing something to determine why it's important to have duplicate JSON sources properties, but I don't think that this is a good practice and I am alerting you about it.

**Comment by remi-stripe** [comment]
@xalhera The idea here is that `source` can be a token id `tok_XXXX`, a card id `card_XXXX` (and a variety of other strings) but also a hash with raw card details like this:
```
{
  "object": "card",
  "number: "4242424242424242",
  "exp_month": 1,
  "exp_year": 2018
}
```

Since both approaches take different types in C# we need separate variables. In the API though they map to the same parameter name: `source`. It's not really a duplicate. You're supposed to only ever send one of those and if you set both the API would just return an error.

**Comment by jaymedavis** [comment]
> Maybe I'm missing something to determine why it's important to have duplicate JSON sources properties, but I don't think that this is a good practice and I am alerting you about it.

I agree. It used to be setup that there was a `StripeSource` (or something like that), but people were trying to send a token and a card. Having the duplicate JSON declarations also causes tools like Swagger to bomb. Instead of having two source json declarations, I think it would be better to go back to a single `Source` object, which has a `Token` and `Card` property. Could throw an exception if more than one is populated. 

It's not a huge deal, but I think it would be a little cleaner this way. There's an issue in this repo somewhere related to this causing issues with Swagger.
